src/Stryker.Core/Stryker.Core.UnitTest/Mutators/CollectionExpressionMutatorTests.cs

```diff
@@ -0,0 +1,523 @@
+                                           .AddSyntaxTrees(syntaxTree);
+        var semanticModel = compilation.GetSemanticModel(syntaxTree);
+
+        var injector = new CodeInjection();
+        var orchestrator = new CsharpMutantOrchestrator(new MutantPlacer(injector),
+                                                        options: new StrykerOptions
+                                                        {
+                                                            MutationLevel    = MutationLevel.Complete,
+                                                            OptimizationMode = OptimizationModes.CoverageBasedTest,
+                                                            ExcludedMutations = Enum.GetValues<Mutator>()
+                                                               .Except([Mutator.CollectionExpression])
+                                                        });
+        syntaxTree = orchestrator.Mutate(syntaxTree, semanticModel);
+        orchestrator.Mutants.Count(a => a.Mutation.Type == Mutator.CollectionExpression).ShouldBe(expectedMutants);
+
+        List<string> references =
+        [
+            typeof(object).Assembly.Location,
+            typeof(List<string>).Assembly.Location,
+            typeof(Enumerable).Assembly.Location,
+            typeof(ImmutableArray<>).Assembly.Location,
+            typeof(ValueType).Assembly.Location,
+            ..Assembly.GetEntryAssembly()?.GetReferencedAssemblies().Select(a => Assembly.Load(a).Location) ?? []
+        ];
+
+        var input = new MutationTestInput
+        {
+            SourceProjectInfo = new SourceProjectInfo
+            {
+                AnalyzerResult = TestHelper.SetupProjectAnalyzerResult(projectFilePath: "/c/project.csproj",
+                                                                       properties: new Dictionary<string, string>
+                                                                       {
+                                                                           { "TargetDir", "" },
+                                                                           { "AssemblyName", "AssemblyName" },
+                                                                           {
+                                                                               "TargetFileName",
+                                                                               "TargetFileName.dll"
+                                                                           }
+                                                                       },
+                                                                       references: references.ToArray()
+                                                                      )
+                                           .Object
+            }
+        };
+
+        var target = new CsharpCompilingProcess(input);
+
+        try
+        {
+            var result =
+                target.Compile([
```

Comment:
Could you explain why you are compiling the mutations here?

Reply:
To ensure that adding a mutation doesn't introduce ambiguity on compile. I've tried to test as many cases as I can that have the compiler infer a collection type from the context, and I don't wont any of them (mainly the mutation to a empty collection) to have it unable to infer a type. I originally added these to the integration project, but I moved them because I was struggling to consistently run the integration tests. I can move it back, or remove it out-right, as these mutations should be implemented as `(mutated ? [original] : [])` and therefore always have the same type information as before.

Reply:
Sounds about right. It would make more sense to me to test these things in the integration tests. But I also don't mind keeping it here, since it makes sense to test the return types for these kind of mutations.